### ISSUE TITLE
Modal: accessibility warning, aria-hidden

### ISSUE THREAD

**Issue by pmlk** [daisyui/react-daisyui#476]
The `<Modal>` component produces the following warning in Chrome when I open it:

<img width="588" alt="Chrome console warning: 'Blocked aria-hidden on an element because its descendant retained focus. The focus must not be hidden from assistive technology users. Avoid using aria-hidden on a focused element or its ancestor. Consider using the inert attribute instead, which will also prevent focus. For more details, see the aria-hidden section of the WAI-ARIA specification at https://w3c.github.io/aria/#aria-hidden.
Element with focus: <dialog.modal>
Ancestor with aria-hidden: <dialog.modal>'" src="https://github.com/user-attachments/assets/5457e8e8-5a94-4c49-baa5-9cfb965a98bc" />

`aria-hidden` is still `true` even though the modal is in focus and visible to seeing people, so it should be accessible to people relying on assistive technologies which it wouldn't be if it wasn't blocked by the browser. `aria-hidden` should be `false` when the modal is open and / or in focus.

For the sake of this issue, I can reproduce this with almost an exact copy from the storybook https://react.daisyui.com/?path=/story/actions-modal--default

```ts
import { useCallback, useRef } from 'react';
import { Button, Modal } from 'react-daisyui';

export default function Foo() {
  const ref = useRef<HTMLDialogElement>(null);
  const handleShow = useCallback(() => {
    ref.current?.showModal();
  }, [ref]);

  return (
    <div>
      <Button onClick={handleShow}>Open</Button>
      <Modal ref={ref} backdrop={true}>
        <p>contents</p>
      </Modal>
    </div>
  );
}
```

## Questions

**Am I doing anything wrong or forgetting something? Should I handle this as a user of the modal component or should the component handle this internally?** I would prefer the latter and I would be happy to create a pull request if that is the desired approach. 

I found [this commit](https://github.com/daisyui/react-daisyui/commit/9e5064ddf991cc3366c0c3fd05a6fda121ed04cf) which looks like it was made in an attempt to toggle the `aria-hidden` attribute based on the `open` prop. By my understanding this falls short in my instance, because I'm not passing in the `open` prop and controlling the modal through it, but rather by interacting with the underlying `<dialog>` node via the forwarded ref.

### Current workaround

```ts
import { useCallback, useRef } from 'react';
import { Button, Modal } from 'react-daisyui';

export default function Foo() {
  const ref = useRef<HTMLDialogElement>(null);
  const handleShow = useCallback(() => {
    ref.current?.showModal();
  }, [ref]);

  const handleFocus = () => {
    if (ref.current) {
      ref.current.ariaHidden = String(!ref.current.open);
    }
  };

  return (
    <div>
      <Button onClick={handleShow}>Open</Button>
      <Modal ref={ref} backdrop={true} onFocus={handleFocus} onBlur={handleFocus}>
        <p>contents</p>
      </Modal>
    </div>
  );
}
```

I found this [SO thread](https://stackoverflow.com/questions/62238716/using-ref-current-in-react-forwardref) which may be useful for the implementation to handle this inside the modal component.
